### PR TITLE
[DUOS-1502][risk=no] Boosting health check test coverage

### PR DESCRIPTION
**Pull request by MiriamAlter**
- All new unit tests for Elastic Search health check
- New test for GCS health check exception case
- New tests for Ontology, Sam, & SendGrid health check configuration exception cases

Issue: https://broadworkbench.atlassian.net/browse/DUOS-1502

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/java/org/broadinstitute/consent/http/health/ElasticSearchHealthCheckTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+
+public class ElasticSearchHealthCheckTest implements WithMockServer {
+    private ElasticSearchHealthCheck healthCheck;
+    private ElasticSearchConfiguration config;
+    private RestClient client;
```
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/health/ElasticSearchHealthCheckTest.java]: This is unused

File: src/test/java/org/broadinstitute/consent/http/health/ElasticSearchHealthCheckTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    private MockServerClient mockServerClient;
+
+    @Rule
+    public MockServerContainer container = new MockServerContainer(IMAGE);
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
```
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/health/ElasticSearchHealthCheckTest.java]: minor nit: `openMocks` could be a static import. We try to follow that pattern for the setup.

File: src/test/java/org/broadinstitute/consent/http/health/ElasticSearchHealthCheckTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    @Test
+    public void testCheckServerFailure() throws Exception {
+        initHealthCheck("green", HttpStatusCodes.STATUS_CODE_SERVER_ERROR);
+        try {
+            healthCheck.check();
+            fail("Expected InternalServerErrorException was not thrown.");
+        } catch (InternalServerErrorException e) {
+            return;
+        }
+    }
```
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/health/ElasticSearchHealthCheckTest.java]: This test should expect a failure case - it simplifies the code quite a bit
```suggestion
    @Test(expected = InternalServerErrorException.class)
    public void testCheckServerFailure() throws Exception {
        initHealthCheck("green", HttpStatusCodes.STATUS_CODE_SERVER_ERROR);
        healthCheck.check();
    }
```

File: src/test/java/org/broadinstitute/consent/http/health/GCSHealthCheckTest.java [hunk 4]
```diff
@@ -44,4 +45,13 @@ public void testBucketMissing() {
...
         assertTrue(result.getMessage().contains("GCS bucket unreachable"));
     }
+
+    @Test
+    public void testException() {
+        doThrow(new RuntimeException()).when(store).getRootBucketWithMetadata();
+
+        HealthCheck.Result result = healthCheck.execute();
+        assertFalse(result.isHealthy());
+        assertTrue(result.getMessage().contains("GCS bucket unreachable"));
```
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/health/GCSHealthCheckTest.java:55]: nit: I generally don't like to test error messages - it doesn't really add anything to the test nor does it test any real logic. The basic health condition is the primary concern here. There are exceptions to this rule, but I don't think this is one of them.

File: src/test/java/org/broadinstitute/consent/http/health/OntologyHealthCheckTest.java [hunk 5]
```diff
@@ -36,11 +36,13 @@ public void setUp() {
     MockitoAnnotations.openMocks(this);
```
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/health/OntologyHealthCheckTest.java]: Looks like this is a case where we didn't import the static in the original code, but we should.

File: src/test/java/org/broadinstitute/consent/http/health/SamHealthCheckTest.java [hunk 6]
```diff
@@ -36,14 +36,16 @@ public void setUp() {
     MockitoAnnotations.openMocks(this);
```
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/health/SamHealthCheckTest.java]: Same here ... and this is relatively recent code. Shame on me.

**Review by rushtong — CHANGES_REQUESTED** [review]
Just a few minor suggestions for improvement, but overall, looks good! See comments inline.

File: src/test/java/org/broadinstitute/consent/http/health/ElasticSearchHealthCheckTest.java (hunk 3 above)
Comment (MiriamAlter) [reply src/test/java/org/broadinstitute/consent/http/health/ElasticSearchHealthCheckTest.java]: I used the syntax I learned in SG, but thanks for showing me a better one.

File: src/test/java/org/broadinstitute/consent/http/health/GCSHealthCheckTest.java (hunk 4 above)
Comment (MiriamAlter) [reply src/test/java/org/broadinstitute/consent/http/health/GCSHealthCheckTest.java:55]: The other tests included messages as well. I thought it made the most sense to continue the pattern.

**Review by rushtong — APPROVED** [review]
Thank you for the changes 👍🏽

**Pull request merged**